pycbc/io/hdf.py

```diff
@@ -960,13 +960,38 @@ def recursively_save_dict_contents_to_group(h5file, path, dic):
         python dictionary to be converted to hdf5 format
     """
     for key, item in dic.items():
-        if isinstance(item, (np.ndarray, np.int64, np.float64, str, bytes, tuple, list)):
+        if isinstance(item, (np.ndarray, np.int64, np.float64, str, int, float,
+                             bytes, tuple, list)):
             h5file[path + str(key)] = item
         elif isinstance(item, dict):
             recursively_save_dict_contents_to_group(h5file, path + key + '/', item)
         else:
             raise ValueError('Cannot save %s type' % type(item))
 
+def load_hdf5_to_dict(h5file, path):
+    """
+    Parameters
+    ----------
+    h5file:
+        h5py file to be loaded as a dictionary
+    path:
+        path within h5py file to load: '/' for the whole h5py file
+
+    Returns
+    -------
+    dic:
+        dictionary with hdf5 file group content
+    """
+    dic = {}
+    for key, item in h5file[path].items():
+        if isinstance(item, h5py._hl.dataset.Dataset):
+            dic[key] = item[()]
+        elif isinstance(item, h5py._hl.group.Group):
```

Comment:
This is raising some more Codeclimate issues. Is it possible to just use `h5py.Group` and `h5py.Dataset` here? Or is there a specific reason to use those in `_hl.dataset`?

Reply:
I just used the output I got from doing `type(h5file[path])` for groups and datasets but I checked and `h5py.Group` and `h5py.Dataset` are valid types too, so thanks for solving this

Reply:
Ok nice, but make sure it works! They may be slightly different derived classes.

Reply:
I've checked it with ipython and it seems to work correctly